sway-core/src/sway.pest

```diff
@@ -187,7 +186,7 @@ asm_registers            =  {"(" ~ (asm_register_declaration ~ ("," ~ asm_regist
 asm_register_declaration =  {ident ~ (":" ~ expr)?}
 asm_op                   =  {opcode ~ (asm_immediate|asm_register)* ~ ";"}
 asm_register             =  {ident}
-asm_immediate            =  {"i" ~ u64_integer}
+asm_immediate            =  {"i" ~ basic_integer ~ "u64"?}
```

Comment:
same here, all asm immediate sizes are a bit different and can't be assumed as `u64`. They get parsed into `u6`, `u12`, `u18`, and `u24` based on the instruction size. 
```suggestion
asm_immediate            =  {"i" ~ basic_integer }
```

Reply:
Same as above. Will address the issue.

Reply:
Fixed.